src/Tooltip/main.vala

```diff
@@ -0,0 +1,17 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+public void main () {
+    var button = (Gtk.Button) workbench.builder.get_object ("button");
+
+    button.query_tooltip.connect ((button, _x, _y, _keyboard_mode, tooltip) => {
```

Comment:
Unused parameters in Vala do not need to be prefixed with an underscore, and most demos do not prefix them, so I think it would be better to remove it.
```suggestion
    button.query_tooltip.connect ((button, x, y, keyboard_mode, tooltip) => {
```